pkg/pulp_client/task.go

```diff
@@ -32,6 +34,24 @@ func (r pulpDaoImpl) GetTask(taskHref string) (zest.TaskResponse, error) {
 	return *task, nil
 }
 
+func (r pulpDaoImpl) CancelTask(taskHref string) (zest.TaskResponse, error) {
+	canceled := string(zest.STATESENUM_CANCELED)
+	task, httpResp, err := r.client.TasksAPI.
+		TasksCancel(r.ctx, taskHref).
+		PatchedTaskCancel(zest.PatchedTaskCancel{State: &canceled}).
+		Execute()
+	if err != nil {
+		return zest.TaskResponse{}, err
+	}
+	if httpResp.StatusCode == http.StatusConflict {
+		log.Logger.Warn().Msg("Status Conflict")
+		return *task, nil
+	}
+	defer httpResp.Body.Close()
```

Comment:
should this defer be before the StatusConflict check?

Reply:
yes, good catch